services/oauth/v2/types.go

```diff
@@ -0,0 +1,120 @@
+package v2
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	rudderSync "github.com/rudderlabs/rudder-go-kit/sync"
+	"github.com/rudderlabs/rudder-server/services/controlplane/identity"
+)
+
+type (
+	RudderFlow string
+	AuthType   string
+	ContextKey string
+)
+
+type expirationDate struct {
+	ExpirationDate string `json:"expirationDate"`
+}
+
+// AccountSecret is the access token returned by the oauth server
+type AccountSecret struct {
+	ExpirationDate string          `json:"expirationDate"`
+	Secret         json.RawMessage `json:"secret"`
+}
+type OAuthHandler struct {
+	TokenProvider
+	Logger                    logger.Logger
+	RudderFlowType            RudderFlow
+	CpConn                    ControlPlaneConnector
+	AuthStatusUpdateActiveMap map[string]bool // Used to check if a authStatusInactive request for a destination is already InProgress
+	Cache                     Cache
+	CacheMutex                *rudderSync.PartitionRWLocker
+	ExpirationTimeDiff        time.Duration
+	ConfigBEURL               string
+	LoggerName                string
+}
+type CacheKey struct {
+	WorkspaceID string
+	AccountID   string
+}
+
+type TokenProvider interface {
+	Identity() identity.Identifier
+}
+
+type ControlPlaneRequest struct {
+	Body           string
+	ContentType    string
+	Url            string
+	Method         string
+	destName       string
+	RequestType    string // This is to add more refined stat tags
+	BasicAuthUser  identity.Identifier
+	rudderFlowType RudderFlow
+}
```

Comment:
```suggestion
type ControlPlaneRequest struct {
	body           string
	contentType    string
	url            string
	method         string
	destName       string
	requestType    string // This is to add more refined stat tags
	basicAuthUser  identity.Identifier
	rudderFlowType RudderFlow
}
```

Reply:
Why do we need to expose the variables when all the files belong to the same package

Reply:
We are using these values in `mock_cp_connector.go`.